pkg/promtail/targets/file/tailer.go

```diff
@@ -125,8 +127,6 @@ func (t *tailer) run() {
 			if err := t.handler.Handle(model.LabelSet{}, line.Time, line.Text); err != nil {
 				level.Error(t.logger).Log("msg", "error handling line", "path", t.path, "error", err)
 			}
-		case <-t.quit:
-			return
```

Comment:
Removing the quit channel and instead we will quit when the underlying tailer closes the t.tail.Lines channel above.